### PR TITLE
feat(sdk): support generating openAPI spec

### PR DESCRIPTION
**Pull request by jialeicui**
## Description

`swcli build model` in mnist demo, it will generate spec in `src/svc.yaml`, the content is:

```yaml
info:
  description: StarWhale API
  title: StarWhale API
  version: 1.0.0
openapi: 3.0.0
paths:
  /ppl:
    post:
      requestBody:
        content:
          multipart/form-data:
            schema:
              properties:
                data:
                  format: binary
                  type: string
              required:
              - data
              type: object
        description: starwhale builtin model serving specification for type image
      responses:
        '200':
          content:
            application/json:
              schema:
                properties:
                  data:
                    type: string
                type: object
          description: OK
```

online demo
* [default](https://petstore.swagger.io/?url=https://gist.githubusercontent.com/jialeicui/566d7ebcab2700a5295d965119318684/raw/d1c2d7682e07c37ff710d3b1d2f64c10e615a13b/model-serving-spec-demo.yaml#/)
* [custom](https://petstore.swagger.io/?url=https://gist.githubusercontent.com/jialeicui/566d7ebcab2700a5295d965119318684/raw/57ec409b3c751bcd27a8a12b110154f77ad81bab/model-serving-spec-demo.yaml)

[asciicast](https://asciinema.org/a/fb19YpYSxVkomWAkFY3wNseP8)

## Modules
- [ ] UI
- [ ] Controller
- [ ] Agent
- [ ] Client
- [x] Python-SDK
- [ ] Others

## Checklist
- [x] run code format and lint check
- [x] add unit test
- [ ] add necessary doc

### CODE REVIEW

File: client/starwhale/api/_impl/service.py [hunk 1]
```diff
@@ -1,72 +1,171 @@
...
-        return json.dumps(resp).encode("utf8")
+    @classmethod
+    def from_flask_request(cls, request: flask.Request) -> "Request":
+        body = request.data
+        if not body:
+            # try uploaded files
+            files = list(request.files.values())
+            if len(files) > 0:
+                # support only one file for now
+                body = files[0].stream.read()
```
[the remark this answers is not in the document]
Comment (jialeicui) [reply client/starwhale/api/_impl/service.py:41]: * Common scenarios require only one input object such as an image, a binary file
* If we support multiple files, we need a better abstraction for this `Request` class (we need more work to do)
  * easy to use and keep simple (e.g. flask has request.data, request.json, request.files, request.form)
  * support multiple web framework, e.g. starlette
* Users can use a custom `Input` and JSON requests to walkaround the limitation, e.g.
  * base64.encode the multiple files and merge them into diff fields of the request JSON
  * Extract and decode the files in `CustomInput(Input)`

[the remark this answers is not in the document]
Comment (jialeicui) [reply client/starwhale/api/_impl/service.py:41]: I'll raise an exception when getting multiple files for now and create an issue tracking the refactor of `Request`

File: client/starwhale/core/model/model.py [hunk 2]
```diff
@@ -229,21 +236,26 @@ def _get_service(module: str, pkg: Path) -> Service:
                 svc = v
         if attr:
             cls = getattr(m, attr)
-            # TODO: refine this ugly ad hoc
-            context_holder.context = Context(
-                Path("."), version="-1", project="tmp-project-for-build"
-            )
             ins = cls()
             apis.update(ins.svc.apis)
```
Comment (goldenxinxing) [inline client/starwhale/core/model/model.py:240]: Only the default ppl‘s svc is not empty, add a judgment?

File: client/starwhale/api/_impl/service.py (hunk 1 above)
[the remark this answers is not in the document]
Comment (jialeicui) [reply client/starwhale/api/_impl/service.py:41]: exception added.

File: client/starwhale/api/_impl/service.py [hunk 3]
```diff
@@ -1,72 +1,174 @@
 import json
 import typing as t
+import functools
+from abc import ABC, abstractmethod
 from dataclasses import dataclass
 
-from typing_extensions import Protocol, runtime_checkable
+import flask
 
+from starwhale.base.spec.openapi.components import Schema, OpenApi
```
Comment (tianweidut) [inline client/starwhale/api/_impl/service.py:9]:
```
from starwhale.base.spec.openapi.components import Schema, OpenApi, Response as OpenApiResponse, MediaType, SpecComponent
```
merge imports?

**Review by tianweidut — COMMENTED** [review]
Others LGTM

File: client/starwhale/api/_impl/service.py (hunk 3 above)
Comment (jialeicui) [reply client/starwhale/api/_impl/service.py]: Done, detail: https://github.com/star-whale/starwhale/pull/1590/commits/7fb4d581fee73292a975b9212f11f34b24a0714c

**Review by goldenxinxing — APPROVED** [review]
LGTM

**Pull request merged**